src/bin/main.rs

```diff
@@ -2,13 +2,23 @@ use std::env;
 use std::fs;
 
 use main_error::MainError;
+use serde::{Deserialize, Serialize};
+use tf_demo_parser::demo::header::Header;
+use tf_demo_parser::demo::parser::analyser::MatchState;
 use tf_demo_parser::demo::parser::player_summary_analyzer::PlayerSummaryAnalyzer;
 pub use tf_demo_parser::{Demo, DemoParser, Parse, ParseError, ParserState, Stream};
 
 #[cfg(feature = "jemallocator")]
 #[global_allocator]
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct JsonDemo {
+    header: Header,
+    state: MatchState,
```

Comment:
please [flatten](https://serde.rs/attr-flatten.html) this so it doesn't break the compatibility of the json struct